srcStatic/IniModuleLoader.cpp

```diff
@@ -54,6 +54,15 @@ bool IniModuleLoader::UnloadModules()
 	return result;
 }
 
+std::string IniModuleLoader::GetModuleName(std::size_t index)
+{
+	if (index >= modules.size()) {
+		return "";
```

Comment:
Perhaps a `throw` would be more appropriate here. As per the `std::string` return value, this code is strictly C++ anyway. The translation to a return value for the C interface of the DLL could be done at a higher layer.

Reply:
I don't see much use for the throw here. I would just be catching and translating into an empty string and I don't have a use case for the exception yet. Seems like extra work for no reason. Don't see anything wrong with a throw/catch solution though. If it is important to you, let me know and I'll implement.

Reply:
Sure, there's not much use, and that's largely because of the restrictions imposed by the C API, which is the only place that calls this code. Effectively it's pushing a C way of handling errors (with return codes) onto C++ code, that would normally use exceptions to report errors. It kind of feels like it pollutes the C++ code with old style C-isms.

I suppose you can apply the [YAGNI principle](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it) to justify keeping it as is. Not really a big deal I suppose.